tools/infrastructure/api_compare.py

```diff
@@ -0,0 +1,368 @@
+        print rpc_color + func_name + colors.ENDC
+        print colors.BOLD + "\nMobile API: " + colors.ENDC
+        print "Parameters quantity: ", len(mob_rpc_params)
+        print "Parameters list: ", sorted(mob_rpc_params.keys())
+        # print "\n"
+        print colors.BOLD + "HMI API: " + colors.ENDC
+        print "Parameters quantity: ", len(hmi_rpc_params)
+        print "Parameters list: ", sorted(hmi_rpc_params.keys())
+        print "\n"
+        print("{}Parameters absent in Mobile APIs: {}{}".
+              format(colors.WARN, absent_params_mob_api, colors.ENDC))
+        print("{}Parameters absent in HMI APIs: {}{}".
+              format(colors.WARN, absent_params_hmi_api, colors.ENDC))
+        print "\n"
+        if len(absent_params_mob_api) > 0:
+            FillSummaryResult(
+                func_name, "Missed in Mobile APIs:", absent_params_mob_api
+            )
+        if len(absent_params_hmi_api) > 0:
+            FillSummaryResult(
+                func_name, "Missed in HMI APIs:", absent_params_hmi_api
+            )
+        print "Parameters to check: ", common_parameters_list, "\n"
+        for param in common_parameters_list:
+            mob_items = sorted(mob_rpc_params[param].items())
+            hmi_items = sorted(hmi_rpc_params[param].items())
+            print colors.UNDERLINE + colors.BOLD + param + colors.ENDC, ":"
+            print "In Mobile API :\t", dict(mob_items)
+            print "In HMI API :\t", dict(hmi_items)
+            diff_res = CompareParams(param, mob_items, hmi_items)
+            area = "Attributes difference:"
+            if len(diff_res) > 0:
+                FillSummaryResult(func_name, area, diff_res)
+
+
+summary_color = colors.UNDERLINE + colors.BOLD + colors.BLUE
+
+if args.output == "console":
+    print summary_color + "\nSUMMARY COMPARISON RESULT" + colors.ENDC, ":\n"
+    for rpc_name in sorted(summary_result.keys()):
+        print "\n" + "---" * 60
+        print colors.HEADER + rpc_name + colors.ENDC
+        for problematic_item in summary_result[rpc_name]:
+            item = summary_result[rpc_name][problematic_item]
+            if len(item) > 0:
+                print colors.UNDERLINE + problematic_item + colors.ENDC
+            if type(item) is not dict:
+                print("{}{}{}".format(colors.WARN, item, colors.ENDC))
+            elif type(item) is dict:
+                for param in item.keys():
+                    item_print = colors.UNDERLINE + param + colors.ENDC
+                    print "{} {}".format("Parameter name: ", item_print)
+                    res_val = item[param]
+                    for key in res_val:
+                        print key, ":", colors.FAIL, res_val[key], colors.ENDC
+
+json_summary_result = json.dumps(
+    summary_result, sort_keys=True, indent=4, separators=(',', ': '))
+j = json.loads(json_summary_result)
+xml_summary_result = json2xml(j)
```

Comment:
@pestOO We don't need generate xml all the time. But we need to generate json all the time because xml can be generated only from json format.